internal/keepers/service.go

```diff
@@ -161,9 +170,28 @@ func (s *onDemandUpkeepService) start() {
 			for {
 				select {
 				case log := <-ch:
-					// TODO: add delete method to cache; hack sets a new status and 1
-					// nanosecond expire time
-					s.stateCache.Set(string(log.Key), types.Performed, 1)
+					// TODO: check perform log confirmations; expect the log
+					// provider to send a log again at a higher block number
+					// which will have more confirmations
```

Comment:
can we move this logic outside this block so that we only get logs post confirmation? I would expect it to be a config sent to log poller upon initialisation ?

Reply:
That depends on where we put the confirmations config param. If we use the offChainConfig, that config is only available after plugin initialization which would isolate its value to the plugin. If we use the node config, we can apply the confirmation filter to the logs before they reach the plugin.

Reply:
ccip also uses this as offchain config we can look at their solution https://github.com/smartcontractkit/chainlink-ccip/blob/ccip-develop/core/services/ocr2/plugins/ccip/offchain_config.go#L7-L10